src/frontend/src/handler/alter_source_column.rs

```diff
@@ -147,6 +144,20 @@ pub fn alter_definition_add_column(definition: &str, column: ColumnDef) -> Resul
     Ok(stmt.to_string())
 }
 
+/// FIXME: perhapts we should use sth like `ColumnIdGenerator::new_alter`,
+/// However, the `SourceVersion` is problematic: It doesn't contain `next_col_id`.
+/// (But for now this isn't a large problem, since drop column is not allowed for source yet..)
+///
+/// Besides, the logic of column id handling is a mess.
+/// In some places, we use `ColumnId::placeholder()`, and use `col_id_gen` to fill it at the end;
+/// In other places, we create column id ad-hoc.
+pub fn max_column_id(columns: &Vec<ColumnCatalog>) -> ColumnId {
+    // XXX: should we check the column IDs of struct fields here?
+    columns
+        .iter()
+        .fold(ColumnId::new(i32::MIN), |a, b| a.max(b.column_id()))
```

Comment:
Shall we use `1` instead?

https://github.com/risingwavelabs/risingwave/blob/b05d750671ab848129bcea0a12c7efc3acb37a7e/src/frontend/src/handler/create_table.rs#L113